ui/app/AppLayouts/Chat/ChatColumn.qml

```diff
@@ -118,11 +121,39 @@ StackLayout {
             }
        }
 
-
         ProfilePopup {
             id: profilePopup
+            onBlockButtonClicked: {
+                blockContactConfirmationDialog.contactName = name
+                chatColumnLayout.contact = address
+                blockContactConfirmationDialog.open()
+            }
+            onRemoveButtonClicked: {
+                chatColumnLayout.contactToRemove = address
+                removeContactConfirmationDialog.open()
+            }
         }
 
+        BlockContactConfirmationDialog {
+            id: blockContactConfirmationDialog
+            onBlockButtonClicked: {
+                chatsModel.blockContact(blockContactConfirmationDialog.contactAddress)
+                blockContactConfirmationDialog.close()
+                profilePopup.close()
+            }
+        }
+
+        ConfirmationDialog {
+          id: removeContactConfirmationDialog
+          title: qsTrId("remove-contact")
```

Comment:
Please add a comment with the original English string above. It helps us and future translators understand the context
It's something like:
```
//% Original string
```